man/fi_av.3.md

```diff
@@ -328,6 +348,16 @@ that calls to fi_av_insert following a call to fi_av_remove always reference a
 valid buffer in the fi_addr parameter.  Otherwise it may be difficult to
 determine what the next assigned index will be.
 
+If the address vector is configured with authorization keys, the fi_addr
+parameter cannot be NULL and is used to define the authorization key associated
+with the address being inserted. Acceptable inputs are the fi_addr_t's generated
+from `fi_av_insert_auth_key`. Only `fi_addr[0]` can be used to pass in an
+authorization key fi_addr_t. Return returned fi_addr_t's will map to endpoint
+address against the specified authorization key. These fi_addr_t's can be
+used as the target for local data transfer operations. If the endpoint
+supports `FI_DIRECTED_RECV`, these fi_addr_t's can be used to restrict
+recieve buffers to a specific endpoint address and authorization key.
```

Comment:
Hmm... "if the address vector is configured with authorization keys"...

That brings up that I think we'll want a way for the app to clearly indicate that it wants to use auth_keys, probably as part of AV creation.  Based on what's been discussed so far, setting the fi_domain_attr::auth_key_size == FI_AV_AUTH_KEY would work.  (As defined, we're determining this dynamically based on whether the app calls fi_av_insert_auth_key prior to fi_av_insert).

Adding that requirement basically removes the need to set the ep attribute

Reply:
Originally, I thought about fi_domain_attr::auth_key_size == FI_AV_AUTH_KEY. The challenge I ran into was how to support fi_mr_attr::auth_key and fi_mr_attr::auth_key_size. Specifically, the following behavior.

_Indicates the key to associate with this memory registration. Authorization keys are used to limit communication between endpoints. Only peer endpoints that are programmed to use the same authorization key may access the memory region. The domain authorization key will be used if the auth_key_size provided is 0. This field is ignored unless the fabric is opened with API version 1.5 or greater._

Thoughts on how to handle fi_mr_attr::auth_key_size?

Reply:
I think I would go with:

fi_mr_attr::auth_key_size = sizeof(fi_addr_t)
fi_mr_attr::auth_key = fi_addr_t

Since the domain is set to FI_AV_AUTH_KEY, we can expect the fi_mr_attr fields to align.

Reply:
Since a domain could have multiple AVs, do we need a way to bind the MR to the AV?

Reply:
I'll update the PR with `struct fi_mr_auth_key`